crates/api_models/src/payments.rs

```diff
@@ -3141,11 +3143,12 @@ pub struct PaymentLinkDetails {
     pub pub_key: String,
     pub client_secret: String,
     pub payment_id: String,
-    #[serde(with = "common_utils::custom_serde::iso8601")]
-    pub expiry: PrimitiveDateTime,
+    #[serde(with = "common_utils::custom_serde::iso8601::option")]
+    pub expiry: Option<PrimitiveDateTime>,
     pub merchant_logo: String,
     pub return_url: String,
-    pub merchant_name: crypto::OptionalEncryptableName,
-    pub order_details: Vec<pii::SecretSerdeValue>,
```

Comment:
why are these changed to normal types instead of secret?

Reply:
You can expose when injecting in the html, these must be secret values. If these are normal types then these will be visible in the logs which we do not want